homeassistant/components/synology_dsm/update.py

```diff
@@ -0,0 +1,84 @@
+"""Support for Synology DSM update platform."""
+from __future__ import annotations
+
+from typing import Any, Final
+
+from synology_dsm.api.core.upgrade import SynoCoreUpgrade
+
+from homeassistant.components.update import UpdateEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import EntityCategory
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+from . import SynoApi, SynologyDSMBaseEntity
+from .const import (
+    COORDINATOR_CENTRAL,
+    DOMAIN,
+    SYNO_API,
+    SynologyDSMUpdateEntityEntityDescription,
+)
+
+UPDATE_ENTITIES: Final = [
+    SynologyDSMUpdateEntityEntityDescription(
+        api_key=SynoCoreUpgrade.API_KEY,
+        key="update",
+        name="DSM Update",
+        entity_category=EntityCategory.CONFIG,
+    )
+]
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up demo update entities."""
+    data = hass.data[DOMAIN][entry.unique_id]
+    api: SynoApi = data[SYNO_API]
+    coordinator = data[COORDINATOR_CENTRAL]
+
+    async_add_entities(
+        [
+            SynoDSMUpdateEntity(api, coordinator, description)
+            for description in UPDATE_ENTITIES
+        ]
+    )
+
+
+class SynoDSMUpdateEntity(SynologyDSMBaseEntity, UpdateEntity):
+    """Mixin for update entity specific attributes."""
+
+    entity_description: SynologyDSMUpdateEntityEntityDescription
+
+    def __init__(
+        self,
+        api: SynoApi,
+        coordinator: DataUpdateCoordinator[dict[str, dict[str, Any]]],
+        description: SynologyDSMUpdateEntityEntityDescription,
+    ) -> None:
+        """Initialize the Synology DSM binary_sensor entity."""
+        super().__init__(api, coordinator, description)
```

Comment:
It does nothing, this can be skipped?

```suggestion
```

Reply:
hups ... this should already be removed 🙈